api/teams_test.go

```diff
@@ -636,6 +733,17 @@ var _ = Describe("Teams API", func() {
 							Expect(teamServerDB.CreateTeamCallCount()).To(Equal(1))
 						})
 					})
+
+					Context("when passed Generic OAuth credentials", func() {
+						BeforeEach(func() {
+							team.GenericOAuth = &genericOAuth
+						})
+
+						It("updates the Generic OAuth auth for that team", func() {
+							Expect(response.StatusCode).To(Equal(http.StatusCreated))
+							Expect(teamServerDB.CreateTeamCallCount()).To(Equal(1))
```

Comment:
@vito the context here is that the team doesn't exist (line 652). Because the team doesn't exist, `CreateTeam` is called. The `Update*Auth` methods are not called in this case, setting the the team server is relying on the db to set it up correctly when it creates it (in set.go).

This is in contrast to the update tests below that do the check as you described:
```
It("updates the Generic OAuth auth for that team", func() {
  Expect(response.StatusCode).To(Equal(http.StatusCreated))
  Expect(teamServerDB.CreateTeamCallCount()).To(Equal(1))
  Expect(teamDB.UpdateGenericOAuthCallCount()).To(Equal(1))
})
```

So it looks like its just asserting the dependency is being called. How about we rename the test so the intent is clearer? Something like:
```
Context("when passed Generic OAuth credentials", func() {
It("creates the team", func() {
```